ptf

```diff
@@ -133,6 +134,14 @@ be subtracted from the result by prefixing them with the '^' character.  """
             # Parse --interface
             def check_interface(value):
                 try:
+                    sp = ';'
+                    if sp in value:
+                        value, p_info = value.split(sp, 1)
+                        params = p_info.split(sp)
+                        port = params.pop(0)
+                        params.pop()
```

Comment:
what are these 2 lines for?

if `value` is `'device-port@interface;arg=val;arg2=val'`, then after line 140, `params` is `['arg=val', 'arg2=val']`. Why do we pop the first and last element?

Reply:
@vshuper ping on this

Reply:
The first elem pop is important since it is a port number which is expected to be a key and not the part of port info itself.
E.g. `('params', ['0', 'chnl_id=0', 'device_port=0', 'speed=25G', 'conn_id=17', 'if=ens3f1'])` so the first element `'0'` is popped out of the list and used as a key few lines below:
`getattr(namespace, "port_info")[port] = port_cfg`

Reply:
The pop() of the last element is a mistake and will be removed

Reply:
Done

Reply:
```
 >>> value="0-1@veth1;speed=100g-r2;fec=rs;mtu=9192;other_param=some_value"
>>> sp=";"
>>> if sp in value:
...     value, p_info = value.split(sp, 1)
...     params = p_info.split(sp)
...     port = params.pop(0)
...     print(params)
...
['fec=rs', 'mtu=9192', 'other_param=some_value']
>>>
```
This is your code, with the example you gave in the PR description. It seems to me that the speed parameter is lost. Maybe you should consider writing a unit test (https://github.com/p4lang/ptf/blob/master/utests/tests/test.py).

Reply:
The pop() of the first element have been removed